src/transformState.js

```diff
@@ -5,7 +5,21 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  const modifiedState = state;
+
+  actions.forEach(action => {
+    if (action.type === 'addProperties') {
+      Object.assign(modifiedState, action.extraData);
+    } else if (action.type === 'removeProperties') {
+      for (const key of action.keysToRemove) {
+        delete modifiedState[key];
+      }
+    } else if (action.type === 'clear') {
+      for (const key in modifiedState) {
+        delete modifiedState[key];
+      }
+    }
```

Comment:
Replace with switch case construction.

Reply:
Thank you for your remark! I replaced that.